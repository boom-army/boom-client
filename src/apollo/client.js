import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
