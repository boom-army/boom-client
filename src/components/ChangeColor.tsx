import PaletteIcon from '@mui/icons-material/Palette';
import { Stack, Typography } from "@mui/material";
import { ThemeContext } from "../contexts/theme";
import { useState, useContext } from "react";

export const ChangeColor = ({ iconProps, stackProps }: any) => {
  const { theme, setTheme } = useContext(ThemeContext);

  const colors = [
    "#CA2055",
    "#1DA1F2",
    "#FFAD1F",
    "#794BC4",
    "#F45D22",
    "#17BF63",
  ];

  const [currentColor, setCurrentColor] = useState(colors[0]);

  const changeColor = () => {
    let newAccentColor;

    if (colors.indexOf(currentColor) === colors.length - 1) {
      newAccentColor = colors[0];
    } else {
      newAccentColor = colors[colors.indexOf(currentColor) + 1];
    }    

    setTheme({
      ...theme,
      accentColor: newAccentColor,
      pallette: {
        primary: {
          main: newAccentColor,
        }
      }
    });

    setCurrentColor(newAccentColor);
  };

  return (
      <Stack onClick={changeColor} direction="row" {...stackProps} sx={{cursor: "pointer"}}>
        <PaletteIcon sx={iconProps} />
        <Typography variant="body1">Color</Typography>
      </Stack>
  );
};
