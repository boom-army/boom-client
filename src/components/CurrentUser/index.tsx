import { Box, Grid, IconButton } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { useToggleDrawer } from "../../hooks";
import { WalletMultiButton } from "@solana/wallet-adapter-material-ui";

export const CurrentUser = () => {
  const toggleDrawer = useToggleDrawer();
  
  return (
    <>
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Box mr={1} display="flex" alignItems="center" >
        <WalletMultiButton />
          <IconButton
            color="primary"
            onClick={toggleDrawer(true)}
            sx={{ ml: 1 }}
          >
            <MenuIcon />
          </IconButton>
        </Box>
      </Grid>
    </>
  );
};
