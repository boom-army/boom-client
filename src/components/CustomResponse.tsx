import React from "react";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")({
  height: "70vh",
  fontSize: "1.1rem",
  textAlign: "justify",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
  padding: "4rem",
  "@media screen and (max-width: 530px)": {
    fontSize: "1rem",
  },
});

export const CustomResponse = ({ text }: any) => (
  <Wrapper>
    <p>{text}</p>
  </Wrapper>
);
