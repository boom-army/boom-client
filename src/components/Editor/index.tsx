import { AutoFocusPlugin } from "@lexical/react/LexicalAutoFocusPlugin";
import { AutoScrollPlugin } from "@lexical/react/LexicalAutoScrollPlugin";
import { CheckListPlugin } from "@lexical/react/LexicalCheckListPlugin";
import { ClearEditorPlugin } from "@lexical/react/LexicalClearEditorPlugin";
import { ContentEditable } from "@lexical/react/LexicalContentEditable";
import { HashtagPlugin } from "@lexical/react/LexicalHashtagPlugin";
import { LexicalEditor } from "lexical";
import { LinkPlugin } from "@lexical/react/LexicalLinkPlugin";
import { ListPlugin } from "@lexical/react/LexicalListPlugin";
import { OnChangePlugin } from "@lexical/react/LexicalOnChangePlugin";
import { ReactNode, useEffect, useState } from "react";
import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
import { useRef } from "react";

import { RichTextPlugin } from "./plugins/RichTextPlugin";
import { EditorContext } from "./EditorContext";
import { TabFocusPlugin } from "./plugins/TabFocusPlugin";

interface IEditorProps {
  children?: ReactNode;
  hashtagsEnabled?: boolean;
  autoLinkEnabled?: boolean;
  emojisEnabled?: boolean;
  actionsEnabled?: boolean;
  placeholder?: string;
  listMaxIndent?: number;
  initialEditorState?: string;
  isReadOnly?: boolean;
  onChange?: (editorState: string, editorInstance?: LexicalEditor) => void;
}

export const TextEditor = ({
  children,
  hashtagsEnabled = false,
  initialEditorState,
  isReadOnly = false,
  onChange,
}: IEditorProps) => {
  const [editor] = useLexicalComposerContext();
  const [activeEditor, setActiveEditor] = useState(editor);

  const editorStateRef = useRef(null);
  const placeholderComponent = <div>Write something...</div>;
  const scrollRef = useRef(null);

  useEffect(() => {
    editor.setReadOnly(isReadOnly);
  }, []);

  return (
    <EditorContext.Provider
      value={{ initialEditor: editor, activeEditor, setActiveEditor }}
    >
      {children}
      <div className={`editor-container`} ref={scrollRef}>
        <AutoFocusPlugin />
        <ClearEditorPlugin />
        {hashtagsEnabled && <HashtagPlugin />}
        <AutoScrollPlugin scrollRef={scrollRef} />

        <>
          <RichTextPlugin
            contentEditable={<ContentEditable />}
            placeholder={placeholderComponent}
            initialEditorState={initialEditorState}
          />
          <OnChangePlugin
            ignoreInitialChange={false}
            onChange={(editorState) => {
              onChange?.(JSON.stringify(editorState), activeEditor);
              return (editorStateRef.current = editorState);
            }}
          />
          <ListPlugin />
          <CheckListPlugin />
          <LinkPlugin />
          <TabFocusPlugin />
        </>
      </div>
    </EditorContext.Provider>
  );
};
