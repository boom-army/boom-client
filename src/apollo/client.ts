import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
