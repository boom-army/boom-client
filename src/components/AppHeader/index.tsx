import { useCallback, useEffect, useMemo, useContext } from "react";
import base58 from "bs58";
import {
  AppBar,
  Container,
  Grid,
  Toolbar,
} from "@mui/material";
import { CurrentUser } from "../CurrentUser";
import { PUBLIC_ADDRESS, LOGIN_REGISTER } from "../../queries/auth";
import { USER_FOLLOW } from "../../queries/follow";
import { Theme } from "../../contexts/theme";
import { UserContext } from "../../contexts/user";
import { useMutation } from "@apollo/client";
import { useSnackbar } from "../../contexts/snackbar";
import { useWallet } from "@solana/wallet-adapter-react";
import { FeedDocument } from "../../generated/graphql";
import { Box } from "@mui/system";
import { Link } from "react-router-dom";
import { RoutePath } from "../../constants";

export const AppHeader = () => {
  const { connected, wallet, publicKey, signMessage } = useWallet();
  const { enqueueSnackbar } = useSnackbar();
  const { setUser } = useContext(UserContext);

  const token = localStorage.getItem("token");
  const headerImg =
    localStorage.getItem(Theme.StorageTag) === Theme.Light
      ? "/assets/boom-logo-dark.png"
      : "/assets/boom-logo-light.png";

  const [getNonce] = useMutation(PUBLIC_ADDRESS);
  const [setLogin] = useMutation(LOGIN_REGISTER, {
    refetchQueries: [{ query: FeedDocument }, { query: USER_FOLLOW }],
    onCompleted({ loginRegister }) {
      if (localStorage) {
        setUser(loginRegister.user);
        localStorage.setItem("token", loginRegister.token);
      }
    },
  });

  const walletPublicKey = useMemo(() => publicKey?.toBase58(), [publicKey]);

  const signin = useCallback(async () => {
    if (!walletPublicKey) {
      enqueueSnackbar("Wallet not connected!", { variant: "error" });
      return;
    }

    if (!signMessage) {
      enqueueSnackbar("Wallet does not support message signing!", {
        variant: "error",
      });
      return;
    }

    try {
      const {
        data: { address },
      } = await getNonce({
        variables: { publicAddress: walletPublicKey },
      });
      if (address.hasPublicAddress) {
        const data = new TextEncoder().encode(address.user.nonce);
        const signature = await signMessage(data);

        await setLogin({
          variables: {
            publicAddress: walletPublicKey,
            signature: base58.encode(signature),
          },
        });
        window.location.reload();
        enqueueSnackbar(
          `Wallet ${walletPublicKey} connected to account. Happy posting.`,
          { variant: "success" }
        );
      } else {
        await setLogin({ variables: { publicAddress: walletPublicKey } });
        enqueueSnackbar(`Wallet ${walletPublicKey} created for account.`, {
          variant: "success",
        });
      }
    } catch (error) {
      enqueueSnackbar(`Error connecting: ${error}`, { variant: "error" });
    }
  }, [signMessage, getNonce, setLogin, walletPublicKey, enqueueSnackbar]);

  useEffect(() => {
    if (wallet && !token && connected) signin();
  }, [wallet, signin, token, connected]);

  // const logout = () => {
  //   localStorage.clear();
  //   window.location.replace('/');
  // };

  return (
    <AppBar
      position="static"
      sx={{
        borderBottom: (t) => `1px solid ${t.palette.divider}`,
      }}
    >
      <Toolbar variant="dense">
        {/* Remove this when we go full width */}
        <Container maxWidth="lg">
          <Grid container>
            <Grid item xs={12}>
              <Box display="flex" justifyContent="space-between">
                <Box mr={1} display="flex" alignItems="center">
                  <Link
                    to={RoutePath.HOME}
                    style={{ display: "flex", alignItems: "center" }}
                  >
                    <img src={headerImg} alt="Boom" height={20} />
                  </Link>
                </Box>
                <Box mt={0.5}>
                  <CurrentUser />
                </Box>
              </Box>
            </Grid>
          </Grid>
        </Container>
      </Toolbar>
    </AppBar>
  );
};
