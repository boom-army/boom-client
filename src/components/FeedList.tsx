import InfiniteScroll from "react-infinite-scroll-component";
import React, { useContext } from "react";
import { ApolloError } from "@apollo/client";
import { Box } from "@mui/system";
import { CustomResponse } from "./CustomResponse";
import { Grid, Link } from "@mui/material";
import { HARKL_ID } from "../utils/utils";
import { Loader } from "./Loader";
import { NewTweet } from "./Tweet";
import { NewMeepsCountQuery, Tweet } from "../generated/graphql";
import { BoomHeroStore } from "./Advertising/BoomHeroStore";
import { ThemeContext } from "../contexts/theme";
import { UserContext } from "../contexts/user";
import { TweetThread } from "./Tweet/TweatThread";

interface Props {
  loading?: boolean;
  error?: ApolloError | undefined | any;
  data: Array<Tweet> | undefined;
  newMeeps?: NewMeepsCountQuery["newMeepsCount"] | undefined;
  refetchData?: () => void;
  refetchCount?: () => void;
  fetchMore: (props: any) => void;
}

export const FeedList: React.FC<Props> = ({
  loading,
  error,
  data,
  newMeeps,
  refetchData,
  refetchCount,
  fetchMore,
}) => {
  const { theme } = useContext(ThemeContext);
  const { user: userData } = useContext(UserContext);

  if (loading)
    return (
      <Box sx={{ marginTop: "1rem" }}>
        <Loader />
      </Box>
    );
  if (error) return <CustomResponse text={error.message} />;

  // logout the user if removed from db
  if (data === undefined) {
    localStorage.removeItem("user");
  }

  const fetchData = () => {
    fetchMore({
      variables: {
        offset: data?.length ?? 0,
      },
    });
  };

  const loadNewMeeps = () => {
    refetchData && refetchData();
    refetchCount && refetchCount();
  };

  return (
    <Grid
      container
      id="scrollBox"
      sx={{
        height: "100vh",
        overflow: "auto",
      }}
    >
      {data?.length && loading ? (
        <Box sx={{ marginTop: "1rem" }}>
          <Loader />
        </Box>
      ) : null}
      {userData && <NewTweet key="feedNewTweet" />}
      {newMeeps && newMeeps > 0 ? (
        <Grid item xs={12}>
          <Box
            display={"flex"}
            justifyContent={"center"}
            py={1.5}
            sx={{ backgroundColor: theme.tertiaryColor2 }}
          >
            <Link onClick={loadNewMeeps} sx={{ cursor: "pointer" }}>
              Show {newMeeps} Meep{newMeeps > 1 ? "s" : null}
            </Link>
          </Box>
        </Grid>
      ) : null}
      {data?.length ? (
        <InfiniteScroll
          dataLength={data?.length}
          next={fetchData}
          hasMore={true}
          scrollableTarget="scrollBox"
          loader={
            loading && (
              <Box sx={{ marginTop: "1rem" }}>
                <Loader />
              </Box>
            )
          }
        >
          {userData?.data?.avatarUpdateAuthority !== HARKL_ID && (
            <Grid item xs={12}>
              <BoomHeroStore userData={userData} />
            </Grid>
          )}
          {data?.length
            ? data?.map((tweet) => (
                <TweetThread key={tweet.id} tweet={tweet as Tweet} />
              ))
            : null}
        </InfiniteScroll>
      ) : (
        <Grid item xs={12}>
          <CustomResponse text="No hero's have meeped." />
        </Grid>
      )}
    </Grid>
  );
};
