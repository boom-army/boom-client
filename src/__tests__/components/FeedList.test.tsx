import React from "react";
import { FeedList } from "../../components/FeedList";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";

const feed: any = [
  {
    id: "cl084xdy815761s0n5omoxoem0",
    text: "testing after changes in input converted to ts",
    tags: [],
    isTweetMine: true,
    commentsCount: 1,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1646139451616",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [
      {
        id: "cl0c774p371312s0n533u0b29k",
        emojiId: "+1",
        skin: 1,
        isMine: true,
        count: 1,
        __typename: "Reaction",
      },
    ],
    __typename: "Tweet",
  },
  {
    id: "cl081ymfz8867s0n59y782hdj",
    text: "🥰",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1646134470431",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "cl06ku5qr9948ban5356ugu9b",
    text: "😍",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1646045242515",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzzg3lm6169405vun5im00fsmq",
    text: "😘6210",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645614021678",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzzg3cb6167829vun53f79gud8",
    text: "96541651498",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645614009618",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzzfgqwv163794vun5uuak56j0",
    text: "🦧9459559",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645612955455",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [
      {
        id: "cl06ktwei9507ban540vb5u3s",
        emojiId: "laughing",
        skin: null,
        isMine: true,
        count: 1,
        __typename: "Reaction",
      },
    ],
    __typename: "Tweet",
  },
  {
    id: "ckzz3oekx135415vun5lvhk9vj2",
    text: "",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645593157329",
    parentTweet: null,
    files: [],
    gif: {
      id: "ckzz3oel8135425vun5paofgpoa",
      title: "Seth Meyers Reaction GIF by Late Night with Seth Meyers",
      fixedHeightUrl:
        "https://media3.giphy.com/media/8YsjVmpIpEjNKlrL3D/200.mp4?cid=e0fc1889v16eclvv8bcqa6e0ipe89jd8j7xlga5q2d9zd5cz&rid=200.mp4&ct=g",
      originalUrl:
        "https://media3.giphy.com/media/8YsjVmpIpEjNKlrL3D/giphy.mp4?cid=e0fc1889v16eclvv8bcqa6e0ipe89jd8j7xlga5q2d9zd5cz&rid=giphy.mp4&ct=g",
      __typename: "Gif",
    },
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzz2d1i6131338vun5o0yby7ho",
    text: "😍😍😍😍😍😍😍😍😍",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645590947549",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzz2cjue130742vun5iomylsop",
    text: "🎃",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645590924662",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
  {
    id: "ckzxxkyda117508vun5mu4l56jp",
    text: "dsfsfdsf",
    tags: [],
    isTweetMine: true,
    commentsCount: 0,
    retweetsCount: 0,
    isRetweet: false,
    tipsCount: "0",
    createdAt: "1645522452478",
    parentTweet: null,
    files: [],
    gif: null,
    nft: null,
    user: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      avatar: "",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      __typename: "User",
    },
    reactions: [],
    __typename: "Tweet",
  },
];

const error = {
  message: "An error occured",
};

describe("Feed List Rendering", () => {
  test("renders correctly when User have no Feed list", () => {
    const tree = renderer
      .create(<FeedList data={undefined} loading={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when User have Feed list", () => {
    const tree = renderer
      .create(<FeedList data={feed} loading={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when Feed list loading", () => {
    const tree = renderer
      .create(<FeedList data={feed} loading={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when error", () => {
    const tree = renderer
      .create(<FeedList data={undefined} error={error} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
