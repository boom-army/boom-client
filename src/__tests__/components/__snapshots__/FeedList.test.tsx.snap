// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed List Rendering renders correctly when Feed list loading 1`] = `
<div
  className="css-zb8ne"
>
  <div
    className="css-15inajd"
  >
    <p>
      Follow some people to get some feed updates
    </p>
  </div>
</div>
`;

exports[`Feed List Rendering renders correctly when User have Feed list 1`] = `
<div
  className="MuiBox-root css-15g2oxy"
>
  <div
    className="MuiBox-root css-1l4w6pd"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1tww5sk-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Feed List Rendering renders correctly when User have no Feed list 1`] = `
<div
  className="MuiBox-root css-15g2oxy"
>
  <div
    className="MuiBox-root css-1l4w6pd"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1tww5sk-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Feed List Rendering renders correctly when error 1`] = `
<div
  className="css-15inajd"
>
  <p>
    An error occured
  </p>
</div>
`;


